Cheat Finder Usability Improvements

PugsyMAME opened the pull request:
1. Changes visual indications to make it nicer & quicker to use.
2. Adds selectable automatic pause capability to the cheat finder menu.
3. Allows the Test/Write poke to be changed
4. Adds another cheat format capability for cheat.simple (write support only)

cuavas commented:
Can people please at least glance at the diff before merging?  This would light up a git diff in red due to the way it has trailing spaces, and tabs following spaces, and it doesn't even line up in the diff view on github due to mixing space and tab indent.  Also, there are a few things that stand out straight away:
* `pokevaltable` is not localisable
* In English, "&" is used in company names, not as a general abbreviation for "and"
* `leftop_text`, `rightop_text`, `expression_text` are assigned strings that aren't localisable
* Lines 564 and 583 have strange ideas about capitalisation
* Line 583 composes a sentence by concatenation and hence isn't properly localisabe (not introduced)
* What's going on at line 655 with the message format?

PugsyMAME commented:
PR #4618 hopefully addresses most of the issues.

Apologies about the whitespace/grammar issues.

The localisable issues are down to my own ignorance that such a thing even existed. I've had a go at fixing them but I may have swapped one issue for another....
